api/v3/Activity.php

```diff
@@ -300,6 +300,13 @@ function _civicrm_api3_activity_get_spec(&$params) {
 function civicrm_api3_activity_get($params) {
   $options = _civicrm_api3_get_options_from_params($params, FALSE, 'Activity', 'get');
   $sql = CRM_Utils_SQL_Select::fragment();
+  if (in_array(['source_contact_name', 'source_contact_name desc', 'source_contact_name asc'], $options['return'])) {
+    $sql->join(
+      'source_contact',
+      '!joinType civicrm_activity_contact !alias ON (!alias.activity_id = a.id AND record_type_id = ' .  CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_ActivityContact', 'record_type_id', 'Activity Source') . ')',
```

Comment:
this isn't right yet

Reply:
needs to be a subquery with min or max around display name I think

Reply:
nope just checked & the UI only supports source contact - which is a one to one so simple join is fine